### PR TITLE
[batch] Make driver parameters configurable in the UI

### PR DESCRIPTION
**Pull request by jigold**
I tested this in my namespace and everything seems to be working. I tried really hard to keep the order of the parameters the same everywhere to avoid putting values in the wrong places (i.e. 7200 for max_instances instead of for standing worker idle time).

### CODE REVIEW

File: batch/batch/driver/instance_collection/job_private.py
Comment (daniel-goldstein) [inline batch/batch/driver/instance_collection/job_private.py:112]: I prefer `max_new_instances_per_autoscaler_loop`

File: batch/batch/driver/instance_collection/job_private.py
Comment (daniel-goldstein) [inline batch/batch/driver/instance_collection/job_private.py:344]: Why did you change the semantics here? I thought this was just a PR to make these values configurable.

File: batch/batch/driver/instance_collection/pool.py
Comment (daniel-goldstein) [inline batch/batch/driver/instance_collection/pool.py:431]: Without a clarifying comment, I could easily see myself changing this back to using `periodically_call` because I don't pick up on the subtlety that `self.autoscaler_loop_period_secs` is subject to change. I would either add an explicit comment here, write something like `periodically_call_dynamic_sleep(lambda: self.autoscaler_loop_period_secs, self.create_instances)`, or make `periodically_call` accept a `Union[int, Callable[[], int]]` and continue using `periodically_call`.

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
> I tried really hard to keep the order of the parameters the same everywhere to avoid putting values in the wrong places (i.e. 7200 for max_instances instead of for standing worker idle time).

Can we find a way to not have to think this hard? I think these should be keyword-only arguments wherever possible so the only place you have to line things up properly is in sql.

File: batch/batch/driver/instance_collection/job_private.py
Comment (jigold) [reply batch/batch/driver/instance_collection/job_private.py:344]: It was originally to make the parameters configurable. But the current behavior wasn't correct as it would spin up a constant number of instances that wasn't configurable (I believe 300 and/ or 20 per user). I can make the change separately, but felt that it was appropriate here in the context of the overall goal to make these parameters configurable. I didn't want it to look like this parameter was configurable for job-private, but in reality it does nothing.

Your call on whether to make a separate PR.

Comment (jigold) [reply batch/batch/driver/instance_collection/job_private.py:345]: Actually, on second thought, the reason this code was like this in the first place was because the job private manager shouldn't care how quickly it ramps up as it's a job per worker. So 300 was the max for GCP rate limits. I think this change is still good though as it will be consistent with the pool.

File: batch/batch/inst_coll_config.py
Comment (daniel-goldstein) [inline batch/batch/inst_coll_config.py:220]: Can we force these to be keyword-only?

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
One last nit

**Comment by jigold** [comment]
Please keep the WIP tag on so I can check the behavior after it merges in case there's a problem.

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**
